daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/Runner.scala

```diff
@@ -298,231 +300,218 @@ class Runner(
         Set.empty,
       )
 
-    // Removing the early return only makes this harder to read.
-    @SuppressWarnings(Array("org.wartremover.warts.Return"))
```

Comment:
Return is the source of all evil :D
In general, return is not compiled as you expect but using exception. 
Better to avoid it at all cost.

Reply:
What exactly do you mean by that?

Reply:
In general you cannot assume that scala return are translate into simple java returns.
A lot of time they are compile using throw/catch